File: apollo-api/src/main/java/com/apollographql/apollo/api/FileUpload.java
package com.apollographql.apollo.api;

import java.io.File;

public final class FileUpload {
    public final String mimetype;
    public final File file;

    public FileUpload(String mimetype, File file) {
        this.mimetype = mimetype;
        this.file = file;
    }
}

File: libraries/apollo-api/src/jvmMain/kotlin/com/apollographql/apollo/api/FileUpload.kt
@file:JvmName("FileUpload")

package com.apollographql.apollo.api

import okio.buffer
import okio.source
import java.io.File

fun File.toUpload(contentType: String): DefaultUpload {
  return DefaultUpload.Builder()
      .content { sink ->
        source().buffer().use { sink.writeAll(it) }
      }
      .contentLength(length())
      .contentType(contentType)
      .fileName(name)
      .build()
}
